File: tracking.py
import re
from pathlib import Path

path = Path(r'c:/Users/Lenovo/Saved Games/index.html')
text = path.read_text(encoding='utf-8')
m = re.search(r'<!-- LOGIC PROGRAM -->\s*<script>([\s\S]*?)</script>\s*<script>\(function\(\)\{', text)
if not m:
    raise SystemExit('Main script block not found')
compile(m.group(1), str(path), 'exec')
print('JavaScript syntax check: OK')

File: tracking.js
const fs = require('fs');
const path = 'c:/Users/Lenovo/Saved Games/index.html';
const html = fs.readFileSync(path, 'utf8');
const match = html.match(/<!-- LOGIC PROGRAM -->\s*<script>([\s\S]*?)<\/script>\s*<script>\(function\(\)\{/);
if (!match) {
  console.error('Main script block not found');
  process.exit(1);
}
new Function(match[1]);
console.log('JavaScript syntax check: OK');
